### PR TITLE
[chore] Fixed Bug: Examples rendering is broken in 1.26 release for array types

### PR DESCRIPTION
**Pull request by jerbly**
In the 1.26 release the markdown rendering of attribute examples is broken for array types.

An example for a `string[]` attribute type should be displayed as an array: `["a", "b"]`. In 1.26 they are shown as multiple examples appropriate for `string` type: `a`; `b`.

This bug is due to the change to the jinja templates now driven by weaver.

I have fixed the jinja template and run the markdown generation.

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [x] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

File: docs/attributes-registry/aws.md [hunk 1]
```diff
@@ -26,30 +26,30 @@ This document defines generic attributes for AWS services.
...
+| `aws.dynamodb.attributes_to_get`              | string[] | The value of the `AttributesToGet` request parameter.                                      | `["lives", "id"]`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```
Comment (joaopgrassi) [inline docs/attributes-registry/aws.md:32]: hum do you know what is up with the trailing spaces here?

File: docs/attributes-registry/tls.md [hunk 2]
```diff
@@ -10,37 +10,37 @@
...
+| ------------------------------ | -------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------- |
+| `tls.cipher`                   | string   | String indicating the [cipher](https://datatracker.ietf.org/doc/html/rfc5246#appendix-A.5) used during the current connection. [1]                                                                                                  | `TLS_RSA_WITH_3DES_EDE_CBC_SHA`; `TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256`                              | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```
Comment (joaopgrassi) [inline docs/attributes-registry/tls.md:15]: I wonder why these spaces come from, makes the diff large 🤔

File: docs/attributes-registry/aws.md (hunk 1 above)
Comment (jerbly) [reply docs/attributes-registry/aws.md:32]: You mean the table padding? This is added by the command `npm run fix:format` in the makefile for `attribute-registry-generation`. If I comment that out I get:
```
| `aws.dynamodb.attribute_definitions` | string[] | The JSON-serialized value of each item in the `AttributeDefinitions` request field. | `["{ \"AttributeName\": \"string\", \"AttributeType\": \"string\" }"]` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
| `aws.dynamodb.attributes_to_get` | string[] | The value of the `AttributesToGet` request parameter. | `["lives", "id"]` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

File: docs/attributes-registry/tls.md (hunk 2 above)
Comment (jerbly) [reply docs/attributes-registry/tls.md:15]: As above... the table padding was added in the previous release. See PR #917

File: docs/attributes-registry/tls.md [hunk 3]
```diff
@@ -10,37 +10,37 @@
...
+| `tls.client.subject`           | string   | Distinguished name of subject of the x.509 certificate presented by the client.                                                                                                                                                     | `CN=myclient, OU=Documentation Team, DC=example, DC=com`                                              | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `tls.client.supported_ciphers` | string[] | Array of ciphers offered by the client during the client hello.                                                                                                                                                                     | `["\"TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384\", \"TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384\", \"...\""]` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```
Comment (lmolkova) [inline docs/attributes-registry/tls.md:27]: nit: would you mind fixing the original values and removing unnecessary `"` from the examples in yaml?

File: templates/registry/markdown/attribute_macros.j2 [hunk 4]
```diff
@@ -20,8 +20,17 @@
 {% macro name_with_link(attribute, attribute_registry_base_url, lineage_attributes) %}[{{name(attribute)}}]({{attribute_registry_base_url}}/{{ find_lineage(attribute.name, lineage_attributes) | split_id | list | reject("eq", "registry")| first | kebab_case }}.md){% endmacro %}
 
 {% macro examples(attribute) %}{%- if attribute.examples %}
+{%- if "[]" in attribute.type and "template" not in attribute.type %}
```
Comment (lmolkova) [inline templates/registry/markdown/attribute_macros.j2:23]: not for this PR: would be nice to have `is_array` filter for attribute.

File: templates/registry/markdown/attribute_macros.j2 [hunk 5]
```diff
@@ -20,8 +20,17 @@
...
 {%- if attribute.examples is sequence %}
-{%- for example in attribute.examples %}{%if loop.first == false %}; {% endif %}`{{ example }}`{%- endfor %}
+{%- set ns = namespace(all_non_sequence=true) %}
+{%- for example in attribute.examples %}{%- if example is sequence %}{%- set ns.all_non_sequence = false %}{%- endif %}{%- endfor %}
```
Comment (lmolkova) [inline templates/registry/markdown/attribute_macros.j2:26]: nit: is it possible to simplify it with [`select` or `selectattr`](https://jinja.palletsprojects.com/en/3.1.x/templates/#jinja-filters.select) `{%- for example in attribute.examples  | selectattr(...) %}`

not sure if it's possible to express `is sequence` there though

File: templates/registry/markdown/attribute_macros.j2 [hunk 6]
```diff
@@ -20,8 +20,17 @@
...
+{%- if ns.all_non_sequence %}`{{ attribute.examples | trim }}`
+{%- else %}{%- for example in attribute.examples %}{%if loop.first == false %}; {% endif %}`{{ example | trim }}`{%- endfor %}
+{%- endif %}
+{%- else %}`[{{ attribute.examples | trim }}]`
+{%- endif %}
+{%- elif attribute.examples is sequence %}
+{%- for example in attribute.examples %}{%if loop.first == false %}; {% endif %}`{{ example | trim }}`{%- endfor %}
```
Comment (lmolkova) [inline templates/registry/markdown/attribute_macros.j2:33]: it seems 
```j2
{%- for example in attribute.examples %}{%if loop.first == false %}; {% endif %}`{{ example | trim }}`{%- endfor %}
```
 is repeated a few times and can be extracted to a macro

**Review by lmolkova — APPROVED** [review]

File: templates/registry/markdown/attribute_macros.j2 (hunk 4 above)
Comment (lmolkova) [reply templates/registry/markdown/attribute_macros.j2:23]: Created https://github.com/open-telemetry/weaver/issues/180

File: templates/registry/markdown/attribute_macros.j2 (hunk 5 above)
Comment (jerbly) [reply templates/registry/markdown/attribute_macros.j2:26]: This is minijinja so doesn't have quite so many features as jinja. However, it does have [select and selectattr](https://docs.rs/minijinja/latest/minijinja/filters/index.html). The logic here though needs to determine if any of the entries in the array are an array - not to filter them. This is the only way to distinguish between a single array example and a list of array examples.

I'll see if there's some other way to express this.

File: templates/registry/markdown/attribute_macros.j2 (hunk 6 above)
Comment (jerbly) [reply templates/registry/markdown/attribute_macros.j2]: Made a macro for this.

File: templates/registry/markdown/attribute_macros.j2 (hunk 5 above)
Comment (jerbly) [reply templates/registry/markdown/attribute_macros.j2]: Found a way to use `select`.

File: docs/attributes-registry/tls.md (hunk 3 above)
Comment (jerbly) [reply docs/attributes-registry/tls.md]: Removed these quotes.

**Review by lmolkova — APPROVED** [review]
Thank you!

**Review by trisch-me — APPROVED** [review]
thanks! interesting that it took long to realise this error in formatting :)

**Pull request merged**